interoperator/pkg/internal/resources/resources.go

```diff
@@ -178,12 +179,20 @@ func (r resourceManager) ComputeExpectedResources(client kubernetes.Client, inst
 }
 
 // SetOwnerReference updates the owner reference for all the resources
-func (r resourceManager) SetOwnerReference(owner metav1.Object, resources []*unstructured.Unstructured, scheme *runtime.Scheme) error {
+// For multi cluster support we cannot use k8s OwnerReference as
+// owner object might be in a different cluster. So use annotations.
+// Not using labels as it was restrictions of field values.
+func (r resourceManager) SetOwnerReference(owner metav1.Object, ownerGVK schema.GroupVersionKind, resources []*unstructured.Unstructured) error {
```

Comment:
Why not replicate the `SfServiceInstance`/`SfServiceBinding` instance  to the target cluster?

#### Pros
1. Ownership of resources from the `SfPlan` template can be determined locally within the target cluster without reference to the central cluster where interoperator is deployed.
1. Regular ownership management for resources created based on the `SfPlan`.
1. Provides a mechanism to horizontal/distributed scaling (if required in the future). Think about the restart times on a landscape with multiple target clusters.
1. Provides a way to reduce the number of watches across clusters (if required in the future)

#### Cons
1. Duplication of the resources between central and target clusters. But duplication will be only in two clusters for any given interoperator resource. But this might be worth it if it enables the horizontal/distributed scaling.
1. Special handling would still be required to manage the ownership for the replicated `SfServiceInstance`/`SfServiceBinding` instances. But this would be limited to the interoperator's resources.